infra/spawn_pipeline.py

```diff
@@ -85,13 +85,19 @@ def main():
         # mapping between an internal ID and the generate ID
 
         task_id = taskcluster.utils.slugId()
-        task_internal_id = task.pop("ID")
-
+        task_internal_id = task.get("ID")
```

Comment:
```suggestion
        task_internal_id = task["ID"]
```

Also nit: don't remove the newline

Reply:
I'm not sure ID  key will be available always in task dictionary. So I used get instead of direct key access. I'll update this